libpod/runtime_ctr.go

```diff
@@ -150,6 +184,20 @@ func (r *Runtime) newContainer(ctx context.Context, rSpec *spec.Spec, options ..
 		return nil, errors.Wrapf(ErrInvalidArg, "unsupported CGroup manager: %s - cannot validate cgroup parent", r.config.CgroupManager)
 	}
 
+	if restore {
+		// Remove information about /dev/shm mount
+		// for new container from imported checkpoint
+		var devShmPos int
+		for i, vol := range ctr.config.Spec.Mounts {
+			if vol.Destination == DevShmPath {
+				devShmPos = i
+			}
+		}
+		ctr.config.Spec.Mounts[devShmPos] = ctr.config.Spec.Mounts[len(ctr.config.Spec.Mounts)-1]
```

Comment:
I think the generator we use for specs has a RemoveMount function we could use here

Reply:
Replaced the code with g.RemoveMount()